docs/examples/kube2iam.rst

```diff
@@ -0,0 +1,246 @@
+
+    terraform {}
+
+    provider "aws" {
+        region = "${var.aws_region}"
+    }
+
+    variable "aws_region" {
+        description = "AWS Region you want to deploy it in"
+        default     = "eu-west-1"
+    }
+
+    variable "cluster_name" {
+        description = "Name of the cluster"
+    }
+
+    variable "instance_iam_role_arn" {
+        description = "ARN of the instance IAM role"
+    }
+
+    resource "aws_iam_role" "test_role" {
+        name = "test_role"
+        path = "/kube2iam_${var.cluster_name}/"
+
+        assume_role_policy = "${data.aws_iam_policy_document.test_role.json}"
+    }
+
+    data "aws_iam_policy_document" "test_role" {
+        statement {
+            sid = "1"
+
+            actions = [
+                "sts:AssumeRole",
+            ]
+
+            principals {
+                type        = "AWS"
+                identifiers = ["${var.instance_iam_role_arn}"]
+            }
+        }
+    }
+
+    resource "aws_iam_role_policy" "test_role_policy" {
+        name = "test_policy"
+        role = "${aws_iam_role.test_role.id}"
+
+        policy = "${data.aws_iam_policy_document.test_role_policy.json}"
+    }
+
+    data "aws_iam_policy_document" "test_role_policy" {
+        statement {
+            sid = "1"
+
+            actions = [
+                "s3:ListAllMyBuckets",
+            ]
+
+            resources = [
+                "*",
+            ]
+        }
+    }
+
+    output "test_role" {
+        value = "${aws_iam_role.test_role.arn}"
+    }
+
+Now you can run this Terraform project the following way:
+
+.. code-block:: bash
+
+    terraform init
+    terraform apply -var cluster_name=example -var region=eu-west-1 -var instance_iam_role_arn=arn:aws:iam::xxxxxxx:role/my-instance-role
+
+
+When you create a role, you need to make sure you deploy it in the correct
+``path`` and also add an assume role policy to it. That assume role policy
+needs to grant access to the role ARN that is attached to the instances.
+In our example Terraform project above we solved that by adding a variable for
+the ``instance_iam_role_arn`` and the ``cluster_name``.
+
+.. warning::
+   Make sure you use the IAM role ARN attached to your worker Kubernetes 
+   instances as input for ``instance_iam_role_arn``. You can retrieve the
+   IAM role ARN through the AWS console.
+   Don't confuse this with the earlier created IAM policy. 
+
+With the output ``test_role``, you can add that ARN as an annotation to 
+your Pod/Deployment/ReplicaSet etc. 
+In the following example we spin up a pod to list all buckets:
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+        name: aws-cli
+        labels:
+            name: aws-cli
+        annotations:
+            iam.amazonaws.com/role: role-arn
+    spec:
+        containers:
+        - name: aws-cli
+          image: fstab/aws-cli
+          command:
+            - "/home/aws/aws/env/bin/aws"
+            - "s3"
+            - "ls"
```

Comment:
In terms of seeing the output, a Pod yaml actually feels a bit weird - something like the following would probably be better:
```
kubectl run -it aws-cli --restart=Never --image=fstab/aws-cli --overrides='{ "metadata": { "annotations": { "iam.amazonaws.com/role": "role-arn" } } }' --command -- /home/aws/aws/env/bin/aws s3 ls
```

Making sure readers know to replace `role-arn` with their value

Reply:
@dippynark for me this example is not really to show the output (which is a nice added value). But more to show people what they have to add to their pod/deployment/...
So I am not entirely sure I want to switch it to an `kubectl run` command.

Reply:
makes sense